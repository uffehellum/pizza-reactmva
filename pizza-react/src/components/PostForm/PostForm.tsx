import React, { Component, FormEvent } from 'react'
import { connect } from 'react-redux'
import { newPost } from '../../redux/actions/postActions'
import { Post } from '../../types'
import SamplePostForm from '../SamplePosts/SamplePostForm';

export class PostForm extends Component<any, any> {

    constructor(props: any) {
        super(props)
        this.state = {
            title: this.props.title,
            body: this.props.body,
            submitText: this.props.submitText
        }
    }
    onConsole = () => {
        console.log("title: ", this.state.title);
    }
    onSub = () => {
        this.setState({submitText: "back"})
    }
    onChange = ({ target }: { target: { name: string, value: string } }) =>
        this.setState({ [target.name]: target.value })

    onSubmit = (e: FormEvent<any>) => {
        e.preventDefault()
        const post = {
            title: this.state.title,
            body: this.state.body,
    
        }
        console.log(post);
        this.props.newPost(post)
    }

    render = () =>
        <div>
            <h1>Add Post</h1>
            <br/>
            <form onSubmit={this.onSubmit} className="formSubmit">
                <div>
                    <label >title</label>
                    <input 
                        className="titleField"
                        name="title" type="text"
                        onChange={this.onChange}
                        value={this.state.title} />
                </div>
                <br />
                <div>
                    <label>body</label>
                    <textarea name="body"
                        onChange={this.onChange}
                        value={this.state.body} />
                </div>
                <br />onSubmit
                <button type="submit">Submit</button>
                <button className="postformSubmitButton" onClick={this.onSub.bind(this)}>{this.props.submitText}</button>

            </form>

        </div>
}


function mapDispatchToProps(dispatch: any, ownProps: OwnProps): DispatchProps {
    return {
        // ...ownProps,
        newPost: (post: Post) => newPost(post)(dispatch)
    }
}

interface DispatchProps {
    newPost: (post: Post) => void
}

interface OwnProps { }

interface StateProps { }

function mapStateToProps(state: myReduxState, ownProps: OwnProps): StateProps {
    return {
        ...ownProps,
        // posts: state.posts
    }
}

interface myReduxState { }

interface initialState {
    posts: Post[]
}

export default connect
    <myReduxState, DispatchProps, OwnProps, initialState>
    (mapStateToProps, mapDispatchToProps)
    (PostForm)
