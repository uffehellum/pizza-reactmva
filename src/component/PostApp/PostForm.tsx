import React, { Component, FormEvent } from 'react'
import { connect } from 'react-redux'
import { newPost } from './actions/postActions'
import { Post } from './actions/types'

class PostForm extends Component<any, any> {

    constructor(props: any) {
        super(props)
        this.state = {
            title: '',
            body: ''
        }
    }

    onChange = ({ target }: { target: { name: string, value: string } }) =>
        this.setState({ [target.name]: target.value })

    onSubmit = (e: FormEvent<any>) => {
        e.preventDefault()
        const post = {
            title: this.state.title,
            body: this.state.body,
        }
        console.log(post)
        this.props.newPost(post)
    }

    render = () =>
        <div>
            <h1>Add Post</h1>
            <form onSubmit={this.onSubmit}>
                <div>
                    <label>title</label>
                    <input name="title" type="text"
                        onChange={this.onChange}
                        value={this.state.title} />
                </div>
                <br />
                <div>
                    <label>body</label>
                    <textarea name="body"
                        onChange={this.onChange}
                        value={this.state.body} />
                </div>
                <br />
                <button type="submit">Submit</button>
            </form>
        </div>
}


function mapDispatchToProps(dispatch: any, ownProps: OwnProps): DispatchProps {
    return {
        // ...ownProps,
        newPost: (post: Post) => newPost(post)(dispatch)
    }
}

interface DispatchProps {
    newPost: (post: Post) => void
}

interface OwnProps { }

interface StateProps { }

function mapStateToProps(state: myReduxState, ownProps: OwnProps): StateProps {
    return {
        ...ownProps,
        // posts: state.posts
    }
}

interface myReduxState { }

interface initialState {
    posts: Post[]
}

export default connect
    <myReduxState, DispatchProps, OwnProps, initialState>
    (mapStateToProps, mapDispatchToProps)
    (PostForm)