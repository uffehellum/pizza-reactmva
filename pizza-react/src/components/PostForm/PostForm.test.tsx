import Enzyme, { shallow } from 'enzyme'
import React from 'react';
import Adapter from 'enzyme-adapter-react-16'
//import {PostForm} from '../PostForm/PostForm'
import { PostForm } from './PostForm'
import renderer from 'react-test-renderer';

describe('PostForm without redux', () => {

  beforeAll(() => {
    Enzyme.configure({
      adapter: new Adapter()
    })
  })
  test('testing the component rendered correctly or not', () => {
    const tree = renderer.create(
      <PostForm title="xbox" body="support" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('testsing the postform does it contains Add Post as the header or not', () => {
    const wrapper = shallow(<PostForm title="xbox" body="support" />);
    //console.log("*****************************************")
    console.log(wrapper.find('h1').text());
    expect(wrapper.find('h1').text()).toEqual("Add Post");
  });

  test('testing the postform has Add Post element or not', () => {
    const title = "new post title"
    const body = "new post body"
    const postForm = new PostForm({ title, body });

  });
  it('simulates a button click or not', () => {
    const title = "new post title"
    const body = "new post body"
    const post = {
      title: "new post title",
      body: "new post body"
    }

    const postForm = new PostForm({ title, body });
    postForm.onConsole();
    expect(postForm.state.title).toEqual("new post title");
  });

  it('testing the button simulations with jest and enzyme', () => {
    const title = "new post title"
    const body = "new post body"
    const wrapper = shallow(<PostForm title={title} body={body} submitText={"submit"} />);
    wrapper.find('.postformSubmitButton').simulate('click');
    //console.log(wrapper.state('submitText'));
    const val2 = wrapper.state('submitText')
    expect(val2).toEqual("back");
  })


})
// it('testing the postform on wrapper instance', () => {
//   const wrapper = shallow(<PostForm />);
//   const instance = wrapper.instance();
//   console.log(instance) ;
//   instance.onChange();
//   //wrapper.state.title;

// })
// console.log("the value here at postform is ", postForm.state.title);
    // postForm.onChange({ target: { name: "title", value: "xbox mock" }});
    // console.log("the value here at postform is ", postForm.state.title);