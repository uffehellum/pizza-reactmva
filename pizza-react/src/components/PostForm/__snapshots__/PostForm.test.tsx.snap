// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm without redux testing the component rendered correctly or not 1`] = `
<div>
  <h1>
    Add Post
  </h1>
  <br />
  <form
    className="formSubmit"
    onSubmit={[Function]}
  >
    <div>
      <label>
        title
      </label>
      <input
        className="titleField"
        name="title"
        onChange={[Function]}
        type="text"
        value="xbox"
      />
    </div>
    <br />
    <div>
      <label>
        body
      </label>
      <textarea
        name="body"
        onChange={[Function]}
        value="support"
      />
    </div>
    <br />
    onSubmit
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      className="postformSubmitButton"
      onClick={[Function]}
    />
  </form>
</div>
`;

exports[`PostForm without redux testing the component rendered correctly or not 2`] = `
<div>
  <h1>
    Add Post
  </h1>
  <br />
  <form
    className="formSubmit"
    onSubmit={[Function]}
  >
    <div>
      <label>
        title
      </label>
      <input
        className="titleField"
        name="title"
        onChange={[Function]}
        type="text"
        value="xbox"
      />
    </div>
    <br />
    <div>
      <label>
        body
      </label>
      <textarea
        name="body"
        onChange={[Function]}
        value="support"
      />
    </div>
    <br />
    onSubmit
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      className="postformSubmitButton"
      onClick={[Function]}
    />
  </form>
</div>
`;
